### ISSUE TITLE
watermark for OPENAI API calls?

### ISSUE THREAD

**Issue by ttbuffey** [jwkirchenbauer/lm-watermarking#42]
dear Author, 
If I want to add watermark when I directly call GTP API calls(as shown as below) in product development, is the approach demostrated in paper will also works for API calls? Any suggestion for how I can achieve this?

`POST https://{endpoint}/openai/deployments/{deployment-id}/completions?api-version=2024-06-01

{
 "prompt": [
  "tell me a joke about mango"
 ],
 "max_tokens": 32,
 "temperature": 1.0,
 "n": 1
}`

**Comment by jwkirchenbauer** [comment]
This method is only applicable to models that expose the next token prediction distribution (sometimes referred to as the logits, pre-normalization). Some API models have exposed the scores per token in the vocabulary, or at least the top-k of them, but generally as it presents a security risk and powerful model distillation opportunity, most API's today do not expose these scores.
Generally, this is "first party" (1P) watermarking scheme which requires that the model owner/server runs the watermark during generation. "Third party" 3P schemes have been proposed which require only query access (as you sketch above) to apply the watermark to the final generated text the user receives. I recommend searching for "black-box" or 3P watermarking schemes to learn more about this idea.
